guppylang/checker/linearity_checker.py

```diff
@@ -603,6 +610,16 @@ def is_inout_var(place: Place) -> TypeGuard[Variable]:
     return isinstance(place, Variable) and InputFlags.Inout in place.flags
 
 
+def has_explicit_copy(ty: Type) -> bool:
+    """Checks whether a type has an explicit copy function.
+
+    Currently, this is only the case for arrays with copyable elements."""
+    if not is_array_type(ty):
+        return False
+    elem_ty = get_element_type(ty)
+    return ty.droppable and elem_ty.copyable
```

Comment:
Why is the `ty.droppable` restriction needed?

Reply:
ping

Reply:
I removed it now - it was accidentally left over from before I realised I just cared about the elements being copyable and the array being affine is automatically inferred from that